thread_runner.py

```diff
@@ -30,7 +30,23 @@ def waitFor(self):
     def runLoopedFunction(self):
         while True:
             if self.thread_running:
-                self.output = self.func(**self.args)
+                sleep_counter = 0
+                tries_left = 3
+                while tries_left:
+                    try:
+                        self.output = self.func(**self.args)
+                        tries_left = 0
+                    except Exception as err:
+                        sleep_time = self.DEFAULT_SLEEP_TIME * (10 ** sleep_counter)
+                        logger.warning(f"Error occurred while trying to scrape, retrying after {sleep_time} secs. \
+                                        \n Error: {err}")
+                        time.sleep(sleep_time)
+                        tries_left -= 1
+                        sleep_counter += 1
```

Comment:
You can just use tries_left and make the usage of sleep_counter = (DEFAULT_ATTEMPTS - tries_left) at line 40